Assets/Scripts/BulletManagement/Bullet.cs

```diff
@@ -1,53 +1,56 @@
 ﻿using UnityEngine;
 
-/**
- * Base class for bullets. Bullets are destroyed when they no longer are
- * in the view of the Camera.
- */
-public class Bullet : MonoBehaviour
+namespace BulletManagement
 {
-    public float decayTime = 1f;
-    public float instantiationTime = 0.0f;
+    /**
+    * Base class for bullets. Bullets are destroyed when they no longer are
+    * in the view of the Camera.
+    */
+    public class Bullet : MonoBehaviour
+    {
+        public float decayTime = 1f;
+        public float instantiationTime;
 
-    // Metadata
-    public Emitter emitter = null;
-    public BulletPattern bulletPattern = null;
-    private int bulletId;
+        // Metadata
+        public Emitter emitter;
+        public BulletPattern bulletPattern;
+        private int bulletId;
 
-    void Start()
-    {
-        // Destroy self in N seconds after emitt
-        // Do not do so if set to zero.
-        if (decayTime != 0)
+        private void Start()
         {
-            Destroy(this.gameObject, decayTime);
+            // Destroy self in N seconds after emitt
+            // Do not do so if set to zero.
+            if (decayTime != 0)
+            {
+                Destroy(this.gameObject, decayTime);
+            }
+            instantiationTime = Time.timeSinceLevelLoad;
         }
-        this.instantiationTime = Time.timeSinceLevelLoad;
-    }
 
-    void OnBecameInvisible() {
-        Destroy(this.gameObject);
-    }
+        private void OnBecameInvisible() {
+            Destroy(gameObject);
+        }
 
-    void Update()
-    {
-        float timeAlive = (Time.timeSinceLevelLoad - this.instantiationTime);
+        void Update()
+        {
+            var timeAlive = (Time.timeSinceLevelLoad - instantiationTime);
 
-        // Move in a fixed direction until otherwise
-        // Optimize this call if need be when we hit optimization issues.
-        Vector3 translation = this.bulletPattern.GetTranslation(timeAlive, this.bulletId) * Time.deltaTime;
+            // Move in a fixed direction until otherwise
+            // Optimize this call if need be when we hit optimization issues.
+            Vector3 translation = bulletPattern.GetTranslation(timeAlive, bulletId) * Time.deltaTime;
```

Comment:
I prefer we keep the `this` keyword to make the code easier to read and maintain?

Reply:
I think it actually makes it harder to read since line lengths could grow, and I'm not a big fan of including key words just to include them. But I'm willing to discuss.

Reply:
I am personally okay since our project won't be too large. Usually in a large project I would enforce required namespace and `this` reference even if it seems redundant as code can be come hard to maintain. Please use your best judgement.